source/Locales/ScottLocale/ScottLocalesOfScottDomains.lagda

```diff
@@ -0,0 +1,822 @@
+open import DomainTheory.BasesAndContinuity.Continuity       pt fe 𝓤
+open import DomainTheory.BasesAndContinuity.CompactBasis     pt fe 𝓤
+open import Locales.ScottLocale.Definition                   pt fe 𝓤
+open import DomainTheory.Topology.ScottTopology              pt fe 𝓤
+open import DomainTheory.Topology.ScottTopologyProperties    pt fe 𝓤
+open import Locales.Frame                                    pt fe
+ hiding (∅)
+open import Locales.Compactness                              pt fe
+ hiding (is-compact)
+open import Locales.Spectrality.SpectralLocale               pt fe
+
+open import Locales.SmallBasis pt fe sr
+
+open AllCombinators pt fe
+
+open Locale
+
+open PropositionalTruncation pt hiding (_∨_)
+
+\end{code}
+
+The module:
+
+\begin{code}
+
+open import Locales.ScottLocale.ScottLocalesOfAlgebraicDcpos pt fe 𝓤
+
+\end{code}
+
+contains a proof that the Scott locale of any algebraic dcpo is a spectral
+locale.
+
+In this module, we extend this proof by showing that the Scott locale is
+spectral.
+
+\section{Preliminaries}
+
+The following function expresses a list being contained in a given subset.
+
+\begin{code}
+
+_⊆⊆_ : {𝓤 𝓥 : Universe} {X : 𝓤  ̇} → List X → 𝓟 {𝓥} {𝓤} X → 𝓤 ⊔ 𝓥  ̇
+_⊆⊆_ {_} {_} {X} xs U = (x : X) → member x xs → x ∈ U
+
+\end{code}
+
+We define the following predicate that expresses what it means for two elements
+of a DCPO `𝓓` to be “bounded above”.
+
+\begin{code}
+
+bounded-above : (𝓓 : DCPO {𝓤 ⁺} {𝓤}) → ⟨ 𝓓 ⟩∙ → ⟨ 𝓓 ⟩∙ → Ω (𝓤 ⁺)
+bounded-above 𝓓 x y = ∥ upper-bound (binary-family 𝓤 x y) ∥Ω
+ where
+  open Joins (λ a b → a ⊑⟨ 𝓓 ⟩ₚ b)
+
+infix 30 bounded-above
+
+syntax bounded-above 𝓓 x y = x ↑[ 𝓓 ] y
+
+\end{code}
+
+For the proof of spectrality, we will also need the following assumption
+
+\begin{code}
+
+decidability-condition : (𝓓 : DCPO {𝓤 ⁺} {𝓤}) → 𝓤 ⁺  ̇
+decidability-condition 𝓓 = (c d : ⟨ 𝓓 ⟩∙) →
+                            is-compact 𝓓 c → is-compact 𝓓 d →
+                             is-decidable (bounded-above 𝓓 c d holds)
+
+\end{code}
+
+\section{The proof}
+
+As mentioned previously, we assume a couple of things.
+
+  1. The dcpo `𝓓` in consideration is large and locally small.
+  2. It is pointed.
+  3. It has a specified small compact basis.
+  4. It satisfies the aforementioned decidability condition.
+  5. It is bounded complete (which means it is a Scott domain when combined
+     with the algebraicity condition).
+
+\begin{code}
+
+open import DomainTheory.BasesAndContinuity.ScottDomain pt fe 𝓤
+open DefinitionOfBoundedCompleteness
+
+module SpectralScottLocaleConstruction
+        (𝓓    : DCPO {𝓤 ⁺} {𝓤})
+        (hl   : has-least (underlying-order 𝓓))
+        (hscb : has-specified-small-compact-basis 𝓓)
+        (dc   : decidability-condition 𝓓)
+        (bc   : bounded-complete 𝓓 holds)
+        (pe   : propext 𝓤) where
+
+ open ScottLocaleConstruction 𝓓 hscb pe
+
+\end{code}
+
+We denote by `Σ𝓓` the large and locally small Scott locale of the dcpo `𝓓`.
```

Comment:
I know that the usual notation for the Scott topology of `𝓓` is `Σ𝓓` (and I told you so verbally). However, I think it is very confusing to use `Σ` for this in the context of dependent type theory. So please rename. But also notice that the notation you use in the comments doesn't match the notation you use in the Agda code.